domain/application/state/unit.go

```diff
@@ -1883,8 +1926,10 @@ WHERE     n.uuid = $unitUUID.uuid
 			return errors.Capture(err)
 		}
 		err := tx.Query(ctx, queryUnitPublicAddressesStmt, ident).GetAll(&address)
-		if err != nil {
-			return errors.Errorf("querying public addresses for unit %q: %w", uuid, err)
+		if errors.Is(err, sqlair.ErrNoRows) {
+			return nil
+		} else if err != nil {
+			return errors.Errorf("querying addresses for unit %q (and it's services): %w", uuid, err)
```

Comment:
I'm unsure about not returning a domain error for no addresses here, but still...
```suggestion
		if err != nil && !errors.Is(err, sqlair.ErrNoRows) {
			return errors.Errorf("querying addresses for unit %q (and it's services): %w", uuid, err)
```

Reply:
AMHO when a state method can return a nil list, it should return a nil list. It is up to service layer to decide whether it is wrong or not. Lot more flexible this way.

Reply:
Agree with @gfouillet.

Imo, the 'business logic' fact that units/k8s services must have an address is a service layer concern, the state layer should be agnostic. Sometimes we forget how abstract the state layer is (or rather, should be).

Reply:
Yeah I never said I was sure that an error had to be returned. My comment is on refactoring the two separate ifs into one.

Reply:
But yes, your suggestion is better